menas/ui/service/EntityDialog.js

```diff
@@ -0,0 +1,352 @@
+  }
+
+  onNameChange() {
+    let sName = this.oDialog.getModel("entity").getProperty("/name");
+    if (GenericService.isValidEntityName(sName)) {
+      DatasetService.hasUniqueName(sName, this.oDialog.getModel("entity"))
+    } else {
+      this.oDialog.getModel("entity").setProperty("/nameUnique", true);
+    }
+  }
+
+  onSchemaSelect(oEv) {
+    let sSchemaId = oEv.getParameter("selectedItem").getKey();
+    this.schemaService.getAllVersions(sSchemaId, this.oController.byId("schemaVersionSelect"),
+      this.oDialog.getModel("entity"), "/schemaVersion")
+  }
+
+}
+
+class AddDatasetDialog extends DatasetDialog {
+
+  constructor(oDialog, datasetService, schemaService, oController) {
+    super(oDialog, datasetService, schemaService, oController);
+  }
+
+  onPress() {
+    this.oDialog.setBusy(true);
+    this.oDialog.open();
+
+    this.schemaService.getList(this.oDialog).then(oData => {
+      this.schemaService.getAllVersions(oData[0]._id);
+      let oFirstSchema = this.oDialog.getModel("schemas").oData[0];
+
+      this.oDialog.setModel(new sap.ui.model.json.JSONModel({
+        name: "",
+        description: "",
+        schemaName: oFirstSchema._id,
+        schemaVersion: oFirstSchema.latestVersion,
+        hdfsPath: "/",
+        hdfsPublishPath: "/",
+        isEdit: false,
+        title: "Add"
+      }), "entity");
+
+      this.oDialog.setBusy(false);
+    })
+  }
+
+}
+
+class EditDatasetDialog extends DatasetDialog {
+
+  constructor(oDialog, datasetService, schemaService, oController) {
+    super(oDialog, datasetService, schemaService, oController);
+  }
+
+  onPress() {
+    this.oDialog.setBusy(true);
+    this.oDialog.open();
+
+    this.schemaService.getList(this.oDialog).then(() => {
+      let current = this.oController._model.getProperty("/currentDataset");
+
+      current.isEdit = true;
+      current.title = "Edit";
+      this.schemaService.getAllVersions(current.schemaName, this.oController.byId("schemaVersionSelect"));
+
+      this.oDialog.setModel(new sap.ui.model.json.JSONModel(jQuery.extend(true, {}, current)), "entity");
+      this.oDialog.setBusy(false);
+    });
+  }
+
+}
+
+class SchemaDialog extends EntityDialog {
+
+  constructor(oDialog, schemaService, oController) {
+    super(oDialog, schemaService, oController)
+    oController.byId("newSchemaAddButton").attachPress(this.submit, this)
+    oController.byId("newSchemaCancelButton").attachPress(this.cancel, this)
+    oController.byId("newSchemaName").attachChange(this.onNameChange, this)
+  }
+
+  resetValueState() {
+    this.oController.byId("newSchemaName").setValueState(sap.ui.core.ValueState.None);
+    this.oController.byId("newSchemaName").setValueStateText("");
+  }
+
+  isValid(oSchema) {
+    this.resetValueState();
+
+    let hasValidName = EntityValidationService.hasValidName(oSchema, "Schema",
+      this.oController.byId("newSchemaName"));
+
+    return hasValidName;
```

Comment:
The variable assignment is not needed here.

Reply:
I know, I left it there for readability and to make it easy to just add more validations if necessary.